docs/src/introduction.md

```diff
@@ -0,0 +1,20 @@
+# Introduction
+
+Welcome to "The Book of Steel" &mdash; a guide for a dynamic and versatile programming language
```

Comment:
A nit, could we go with `The Steel Book` instead of the `The Book of Steel`? Happy to discuss this

Reply:
Renamed, it was the first thing that came to my mind to name it, eventually it's up to you